.github/workflows/integration-tests.yaml

```diff
@@ -0,0 +1,186 @@
+---
+name: Integration Tests
+on:
+  pull_request:
+    paths:
+      - "action.yaml"
+      - ".github/workflows/integration-tests.yaml"
+
+jobs:
+  test-default-name:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ./
+        id: job
+      - name: Test job name
+        run: '[[ "${output}" == "${expected}" ]] || exit 1'
+        env:
+          output: ${{ steps.job.outputs.name }}
+          expected: ${{ github.job }}
+      - name: Test job ID
+        run: |
+          [[ "${job_id}" =~ ^[0-9]+$ ]] || exit 1
+        env:
+          job_id: ${{ steps.job.outputs.id }}
+
+  test-custom-name:
+    name: Custom Name
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ./
+        id: job
+      - name: Test job name
+        run: '[[ "${output}" == "${expected}" ]] || exit 1'
+        env:
+          output: ${{ steps.job.outputs.name }}
+          expected: Custom Name
+      - name: Test job ID
+        run: |
+          [[ "${job_id}" =~ ^[0-9]+$ ]] || exit 1
+        env:
+          job_id: ${{ steps.job.outputs.id }}
+
+  test-empty-name:
+    name: ""
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ./
+        id: job
+      - name: Test job name
+        run: '[[ "${output}" == "${expected}" ]] || exit 1'
+        env:
+          output: ${{ steps.job.outputs.name }}
+          expected: test-empty-name
```

Comment:
why this instead `${{ github.job }}` like the other null-y cases?

Reply:
Fair question. I think I'll make the other tests use a literal name and leave `${{ github.job }}` as a comment